src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1553,52 +1536,40 @@ protected virtual SynthesizedAttributeData TrySynthesizeIsByRefLikeAttribute()
             return Compilation.TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_IsByRefLikeAttribute__ctor);
         }
 
-        internal void EnsureIsReadOnlyAttributeExists()
+        private void EnsureEmbeddableAttributeExists(EmbeddableAttributes attribute)
         {
             Debug.Assert(!_needsGeneratedAttributes_IsFrozen);
 
-            if (_needsGeneratedIsReadOnlyAttribute_Value || Compilation.NeedsGeneratedIsReadOnlyAttribute)
+            if ((GetNeedsGeneratedAttributesInternal() & attribute) != 0)
             {
                 return;
             }
 
             // Don't report any errors. They should be reported during binding.
-            if (Compilation.CheckIfIsReadOnlyAttributeShouldBeEmbedded(diagnosticsOpt: null, locationOpt: null))
+            if (Compilation.CheckIfAttributeShouldBeEmbedded(attribute, diagnosticsOpt: null, locationOpt: null))
             {
-                _needsGeneratedIsReadOnlyAttribute_Value = true;
+                SetNeedsGeneratedAttributes(attribute);
             }
         }
 
-        internal void EnsureIsUnmanagedAttributeExists()
+        internal void EnsureIsReadOnlyAttributeExists()
         {
-            Debug.Assert(!_needsGeneratedAttributes_IsFrozen);
-
-            if (_needsGeneratedIsUnmanagedAttribute_Value || Compilation.NeedsGeneratedIsUnmanagedAttribute)
-            {
-                return;
-            }
+            EnsureEmbeddableAttributeExists(EmbeddableAttributes.IsReadOnlyAttribute);
+        }
 
-            // Don't report any errors. They should be reported during binding.
-            if (Compilation.CheckIfIsUnmanagedAttributeShouldBeEmbedded(diagnosticsOpt: null, locationOpt: null))
-            {
-                _needsGeneratedIsUnmanagedAttribute_Value = true;
-            }
+        internal void EnsureIsUnmanagedAttributeExists()
```

Comment:
>EnsureIsUnmanagedAttributeExists [](start = 22, length = 32)

nit: consider inlining this method (only one use as far as I can tell).
Comment may be applicable for other `Ensure...` methods as well.